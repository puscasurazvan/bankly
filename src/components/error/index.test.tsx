import { render, screen } from "@testing-library/react";
import { Error } from ".";

describe("Error component", () => {
  it("renders default error message", () => {
    render(<Error />);
    expect(
      screen.getByText(/Error loading your transaction history/)
    ).toBeInTheDocument();
    expect(screen.getByText(/⚠️/)).toBeInTheDocument();
  });

  it("renders custom error message", () => {
    render(<Error displayText="Something went wrong!" />);
    expect(screen.getByText(/Something went wrong!/)).toBeInTheDocument();
    expect(screen.getByText(/⚠️/)).toBeInTheDocument();
  });
});
