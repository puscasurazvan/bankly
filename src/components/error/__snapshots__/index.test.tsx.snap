// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error component > should render as expected 1`] = `
<DocumentFragment>
  <div
    aria-live="assertive"
    class="error"
    role="alert"
  >
    Error loading your transaction history. ⚠️
  </div>
</DocumentFragment>
`;
